core/common/src/main/java/alluxio/retry/TimeoutRetry.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.retry;
+
+import alluxio.util.CommonUtils;
+
+import com.google.common.base.Preconditions;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * An option which allows retrying based on a maximum retry timeout.
+ */
+@NotThreadSafe
+public class TimeoutRetry implements RetryPolicy {
```

Comment:
nit: since we are using the timeout retry policy with constant sleep periods, we could avoid introducing it and instead use `CountingRetry`